File: new-pc-main/library-seat-manage/app/src/main/java/com/example/libraryseatsmanage/api/ApiService.java
package com.example.libraryseatsmanage.api;

import com.example.libraryseatsmanage.model.ApiResponse;
import com.example.libraryseatsmanage.model.CheckInRequest;
import com.example.libraryseatsmanage.model.LoginRequest;
import com.example.libraryseatsmanage.model.LoginResponse;
import com.example.libraryseatsmanage.model.RegisterRequest;
import com.example.libraryseatsmanage.model.Reservation;
import com.example.libraryseatsmanage.model.ReservationRequest;
import com.example.libraryseatsmanage.model.Seat;
import com.example.libraryseatsmanage.model.UserProfile;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {

    @POST("login")
    Call<ApiResponse<LoginResponse>> login(@Body LoginRequest request);

    @POST("register")
    Call<ApiResponse<Object>> register(@Body RegisterRequest request);

    @POST("logout")
    Call<ApiResponse<Object>> logout();

    @GET("user/profile")
    Call<ApiResponse<UserProfile>> getUserProfile(@Header("x-user-id") int userId);

    @GET("seats")
    Call<ApiResponse<List<Seat>>> getSeats(@Query("floor") String floor);

    @POST("reservations")
    Call<ApiResponse<Reservation>> createReservation(
            @Header("x-user-id") int userId,
            @Body ReservationRequest request
    );

    @GET("reservations/my")
    Call<ApiResponse<List<Reservation>>> getMyReservations(@Header("x-user-id") int userId);

    @POST("reservations/{id}/cancel")
    Call<ApiResponse<Object>> cancelReservation(
            @Header("x-user-id") int userId,
            @Path("id") int reservationId
    );

    @POST("checkin")
    Call<ApiResponse<Object>> checkIn(
            @Header("x-user-id") int userId,
            @Body CheckInRequest request
    );

    @POST("checkout")
    Call<ApiResponse<Object>> checkOut(
            @Header("x-user-id") int userId,
            @Body CheckInRequest request
    );
}

File: library-seat-manage/app/src/main/java/com/example/seatmanage/data/api/ApiService.kt
package com.example.seatmanage.data.api

import retrofit2.Response
import retrofit2.http.*

interface ApiService {

    @POST("auth/login")
    suspend fun login(@Body body: LoginRequest): Response<ApiResponse<TokenInfo>>

    @POST("auth/register")
    suspend fun register(@Body body: RegisterRequest): Response<ApiResponse<TokenInfo>>

    @GET("user/profile")
    suspend fun getProfile(): Response<ApiResponse<ProfileResponse>>

    @PUT("user/profile")
    suspend fun updateProfile(@Body body: Map<String, String>): Response<ApiResponse<Unit>>

    @GET("seats")
    suspend fun getSeats(@Query("floor") floor: String? = null): Response<ApiResponse<List<SeatItem>>>

    @GET("seats/{id}")
    suspend fun getSeat(@Path("id") id: String): Response<ApiResponse<SeatItem>>

    @POST("reservations")
    suspend fun createReservation(@Body body: ReservationRequest): Response<ApiResponse<ReservationItem>>

    @GET("reservations/my")
    suspend fun getMyReservations(): Response<ApiResponse<List<ReservationItem>>>

    @POST("reservations/{id}/cancel")
    suspend fun cancelReservation(@Path("id") id: Int, @Body body: CancelBody = CancelBody()): Response<ApiResponse<Unit>>

    @POST("reservations/checkin")
    suspend fun checkIn(@Body body: CheckInRequest): Response<ApiResponse<CheckInResult>>

    @POST("reservations/checkout")
    suspend fun checkOut(@Body body: CheckOutRequest): Response<ApiResponse<CheckOutResult>>

    @GET("notifications")
    suspend fun getNotifications(
        @Query("type") type: String? = null,
        @Query("page") page: Int = 1,
        @Query("pageSize") pageSize: Int = 20
    ): Response<ApiResponse<NotificationListResponse>>

    @GET("notifications/unread-count")
    suspend fun getUnreadCount(): Response<ApiResponse<UnreadCountResponse>>

    @PUT("notifications/{id}/read")
    suspend fun markAsRead(@Path("id") id: Int): Response<ApiResponse<Unit>>

    @PUT("notifications/read-all")
    suspend fun markAllRead(): Response<ApiResponse<Unit>>

    @DELETE("notifications/{id}")
    suspend fun deleteNotification(@Path("id") id: Int): Response<ApiResponse<Unit>>

    @GET("history")
    suspend fun getHistory(
        @Query("status") status: String? = null,
        @Query("page") page: Int = 1,
        @Query("pageSize") pageSize: Int = 20
    ): Response<ApiResponse<HistoryListResponse>>
}
